Cubical/Modalities/Lex.agda

```diff
@@ -0,0 +1,279 @@
+      ◯-rec = ◯-ind (λ _ → B-mod) f
+
+      ◯-rec-β : (x : A) → ◯-rec (η x) ≡ f x
+      ◯-rec-β = ◯-ind-β (λ _ → B-mod) f
+
+  open ◯-rec
+
+  module ◯-map (f : A → B) where
+    abstract
+      ◯-map : ◯ A → ◯ B
+      ◯-map = ◯-rec idemp λ x → η (f x)
+
+      ◯-map-β : (x : A) → ◯-map (η x) ≡ η (f x)
+      ◯-map-β x = ◯-rec-β idemp _ x
+
+
+open ◯-rec
+open ◯-map
+
+module IsModalToUnitIsEquiv (A : Type ℓ) (A-mod : isModal A) where
+  abstract
+    inv : ◯ A → A
+    inv = ◯-rec A-mod λ x → x
+
+    η-retract : retract η inv
+    η-retract = ◯-rec-β _ _
+
+    η-section : section η inv
+    η-section = ◯-ind (λ _ → ≡-modal idemp) λ x i → η (η-retract x i)
+
+    η-iso : Iso A (◯ A)
+    Iso.fun η-iso = η
+    Iso.inv η-iso = inv
+    Iso.rightInv η-iso = η-section
+    Iso.leftInv η-iso = η-retract
+
+    η-is-equiv : isEquiv (η-at A)
+    η-is-equiv = isoToIsEquiv η-iso
+
+abstract
+  unit-is-equiv-to-is-modal : {A : Type ℓ} → isEquiv (η-at A) → isModal A
+  unit-is-equiv-to-is-modal p = transport (cong isModal (sym (ua (η , p)))) idemp
+
+  retract-is-modal
+    : {A : Type ℓ} {B : Type ℓ′}
+    → (A-mod : isModal A) (f : A → B) (g : B → A) (r : retract g f)
+    → isModal B
+  retract-is-modal {A = A} {B = B} A-mod f g r =
+    unit-is-equiv-to-is-modal (isoToIsEquiv (iso η η-inv η-section η-retract))
+    where
+      η-inv : ◯ B → B
+      η-inv = f ∘ ◯-rec A-mod g
+
+      η-retract : retract η η-inv
+      η-retract b = cong f (◯-rec-β A-mod g b) ∙ r b
+
+      η-section : section η η-inv
+      η-section = ◯-ind (λ _ → ≡-modal idemp) (cong η ∘ η-retract)
+
+
+module LiftFam {A : Type ℓ} (B : A → Type ℓ′) where
+  module M = IsModalToUnitIsEquiv (Type◯ ℓ′) ◯-lex
+
+  abstract
+    ◯-lift-fam : ◯ A → Type◯ ℓ′
+    ◯-lift-fam = M.inv ∘ ◯-map (λ a → ◯ (B a) , idemp)
+
+    ⟨◯⟩ : ◯ A → Type ℓ′
+    ⟨◯⟩ [a] = ◯-lift-fam [a] .fst
+
+    ⟨◯⟩-modal : isModalFam ⟨◯⟩
+    ⟨◯⟩-modal [a] = ◯-lift-fam [a] .snd
+
+    ⟨◯⟩-compute : (x : A) → ⟨◯⟩ (η x) ≡ ◯ (B x)
+    ⟨◯⟩-compute x =
+      ⟨◯⟩ (η x)
+        ≡⟨ cong (fst ∘ M.inv) (◯-map-β _ _) ⟩
+      M.inv (η (◯ (B x) , idemp)) .fst
+        ≡⟨ cong fst (M.η-retract _) ⟩
+      ◯ (B x) ∎
+
+
+
+open LiftFam using (⟨◯⟩; ⟨◯⟩-modal; ⟨◯⟩-compute)
+
+module LiftFamExtra {A : Type ℓ} {B : A → Type ℓ′} where
+  ⟨◯⟩←◯ : ∀ {a} → ◯ (B a) → ⟨◯⟩ B (η a)
+  ⟨◯⟩←◯ = transport (sym (⟨◯⟩-compute B _))
+
+  ⟨◯⟩→◯ : ∀ {a} → ⟨◯⟩ B (η a) → ◯ (B a)
+  ⟨◯⟩→◯ = transport (⟨◯⟩-compute B _)
+
+  ⟨η⟩ : ∀ {a} → B a → ⟨◯⟩ B (η a)
+  ⟨η⟩ = ⟨◯⟩←◯ ∘ η
+
+  abstract
+    ⟨◯⟩→◯-section : ∀ {a} → section (⟨◯⟩→◯ {a}) ⟨◯⟩←◯
+    ⟨◯⟩→◯-section = transport⁻Transport (sym (⟨◯⟩-compute  _ _))
+
+
+module Combinators where
+  private
+    variable
+      ℓ′′ : Level
+      A A′ : Type ℓ
+      B : A → Type ℓ′
+      C : Σ A B → Type ℓ′′
+
+  cong-fun : {f g : (x : A) → B x} → f ≡ g → (x : A) → f x ≡ g x
+  cong-fun α x i = α i x
+
+  pair-ext : {p q : Σ A B} (α : p .fst ≡ q .fst) (β : PathP (λ i → B (α i)) (p .snd) (q .snd)) → p ≡ q
```

Comment:
This is https://github.com/agda/cubical/blob/master/Cubical/Data/Sigma/Properties.agda#L37